lib/cqm/converter/hds_record.rb

```diff
@@ -0,0 +1,128 @@
+
+          patient.data_elements << QDM.const_get(dc_type).new.from_json(dc_fixed_keys.to_json)
+        end
+      end
+
+      # Convert patient characteristic birthdate.
+      birthdate = record.birthdate
+      if birthdate
+        birth_datetime = DateTime.strptime(birthdate.to_s, '%s')
+        patient.data_elements << QDM::PatientCharacteristicBirthdate.new(birth_datetime: birth_datetime)
+      end
+
+      # Convert patient characteristic clinical trial participant.
+      # TODO, Adam 4/1: The Bonnie team is working on implementing this in HDS. When that work
+      # is complete, this should be updated to reflect how that looks in HDS.
+      # patient.data_elements << QDM::PatientCharacteristicClinicalTrialParticipant.new
+
+      # Convert patient characteristic ethnicity.
+      ethnicity = record.ethnicity
+      if ethnicity
+        code = QDM::Code.new(ethnicity['code'], ethnicity['codeSystem'], ethnicity['name'])
+        patient.data_elements << QDM::PatientCharacteristicEthnicity.new(data_element_codes: [code])
+      end
+
+      # Convert patient characteristic expired.
+      expired = record.deathdate
+      if expired
+        expired_datetime = DateTime.strptime(expired.to_s, '%s')
+        patient.data_elements << QDM::PatientCharacteristicExpired.new(expired_datetime: expired_datetime)
+      end
+
+      # Convert patient characteristic payer.
+      insurance_providers = record.insurance_providers
+      insurance_providers.each do |payer|
+        start_time = DateTime.strptime(payer['start_time'].to_s, '%s') if payer['start_time']
+        end_time = DateTime.strptime(payer['end_time'].to_s, '%s') if payer['end_time']
+        relevant_period = QDM::Interval.new(start_time, end_time)
+        if start_time || end_time
+          patient.data_elements << QDM::PatientCharacteristicPayer.new(relevant_period: relevant_period)
+        end
+      end
+
+      # Convert patient characteristic race.
+      race = record.race
+      if race
+        code = QDM::Code.new(race['code'], race['codeSystem'], race['name'])
+        patient.data_elements << QDM::PatientCharacteristicRace.new(data_element_codes: [code])
+      end
+
+      # Convert patient characteristic sex.
+      sex = record.gender
+      if sex
+        code = QDM::Code.new(sex, 'Administrative sex (HL7)')
+        patient.data_elements << QDM::PatientCharacteristicSex.new(data_element_codes: [code])
+      end
+
+      # Convert remaining metadata.
+      patient.birth_datetime = DateTime.strptime(record.birthdate.to_s, '%s') if record.birthdate
+      patient.given_names = record.first ? [record.first] : []
+      patient.family_name = record.last if record.last
+      patient.bundle_id = record.bundle_id if record.bundle_id
+
+      patient
+    end
+
+    # This helper method looks at the current state of cqm-models, and builds a hash of
+    # datatype models and their attributes.
+    def gather_qdm_model_attrs
```

Comment:
This code seems to also exist in `lib/cqm/converter/utils.rb`. Should it be in both places?

Reply:
Good catch! This is something I already fixed in the next iteration of this Gem (QDM -> HDS). At the moment, since this isn't breaking anything, I'd rather leave it as is in this PR, unless you feel strongly about it. The sooner we get this in the better IMO.

Reply:
Yeah, if this is fixed in new code, I'm fine leaving it for now.

Reply:
I fixed this anyways.